### PR TITLE
Transaction History Enhancement - Fetching and Filtering

### PR DESCRIPTION
**Pull request by hsyndeniz**
### Description

This pull request enhances the transaction history feature by introducing the ability to fetch and filter transaction history. Users can now conveniently retrieve transaction information for their wallet address while applying filters such as date, amount, sorting, and paging.

### CODE REVIEW

File: src/shared/services/blockscout.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    };
+
+    axios
+      .request(config)
+      .then(response => {
+        /**
+         * Filter out transactions that are below the minimum amount or above the maximum amount.
+         * This is done because the API does not support filtering by amount.
+         */
+        response.data.result = response.data.result.map((transaction: any) => {
```
Comment (obasilakis) [inline src/shared/services/blockscout.ts:83]: why not filter on the first go?

Comment (hsyndeniz) [reply src/shared/services/blockscout.ts]: I fixed the issue and switched from axios to fetch.

**Review by obasilakis — APPROVED** [review]

File: src/main/wallet/WalletScreen.tsx [hunk 2]
```diff
@@ -15,10 +15,33 @@ import { useThemedStyle } from 'src/shared/hooks/useThemedStyle';
...
+    getAccountTransactions(
+      wallet?.address as string,
+      'desc',
+      1,
+      20,
+      1686782724,
+      Date.now(),
+      'to',
+      50000000000000000,
+      1000000000000000000,
```
Comment (juanmanso) [inline src/main/wallet/WalletScreen.tsx:36]: nitpick: too many arguments make it hard to debug/maintain. We could use one (or even more than one) object to better describe params.

No need to do it now, just putting it out there to discuss it and track it for future tweaks

File: src/shared/services/blockscout.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  start_timestamp?: number,
+  end_timestamp?: number,
+  filter_by?: string,
+  min_amount?: number,
+  max_amount?: number,
+): Promise<TransactionList> => {
+  return new Promise(async (resolve, reject) => {
+    // Get the URL for the API
+    const url =
+      `https://paxos1.colosseum.quaiscan.io/api?module=account&action=txlist&address=${address}` +
```
Comment (juanmanso) [inline src/shared/services/blockscout.ts:55]: question: does it matter if it's `paxos1` or `cyprus`, etc.? Should we have that parametrised?

File: src/main/wallet/WalletScreen.tsx (hunk 2 above)
Comment (hsyndeniz) [reply src/main/wallet/WalletScreen.tsx]: it makes sense. replaced with object.

File: src/shared/services/blockscout.ts (hunk 3 above)
Comment (hsyndeniz) [reply src/shared/services/blockscout.ts]: Yes, getting url from keychain makes things easier.

**Pull request merged**